optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/testgen/CorruptedScoreException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.optaplanner.core.impl.score.director.drools.testgen;
+
+public class CorruptedScoreException extends RuntimeException {
```

Comment:
If this extends IllegalStateException, we can reuse it for any score corruption in the normal ScoreDirectors too.

Should be it CorruptedScoreException or ScoreCorruptionException?
Or CorruptionException in case we use it for shadow variable corruption too?